packages/stream/index.d.ts

```diff
@@ -1,2 +1,2 @@
 export { makeQueue, makeStream, makePipe } from './index.js';
```

Comment:
I'm actually somewhat concerned this circular definition will cause these function to not be types properly.

Reply:
Would it be better to not have `index.d.ts` and not have `"types": "./index.d.ts"` in `package.json`?

Reply:
I'm honestly not sure. I probably would have gone with a fully manual `.d.ts` file that also includes `export declare function`. A test file should alleviate or confirm my concerns

Reply:
Story turns out to be that we need an explicit external `index.d.ts`, an internal `types.d.ts` that is shared by `index.js` and `index.d.ts`, and an external validator package to verify `index.d.ts`. There is some possibility that `index.js` and `index.d.ts` might not agree, but this turns out to be a feature. For example, for the purposes of verifying internal integrity, it’s useful for the internal types to be more rigorous than the external types. The primer argument of pump and prime is almost always undefined and optional in practice, but requires unnecessarily verbose handling if it’s optional internally.

Reply:
I usually end up with a `.d.ts` that are more strict, and the implementation that handles the generic/wide case. Especially in the case of overloads (which can't always be expressed by generics only).